### PR TITLE
Add option to disable status bar component

### LINKED ISSUE #1055: Add option to hide status bar item
There is limited space down there, and this takes up a large chunk of it. No package should be adding stuff to the status bar without a corresponding option to hide it…

### PR DESCRIPTION
**Pull request by BenRussert**
- Closes #1055

### CODE REVIEW

**Comment by BenRussert** [comment]
I'll rethink and see if I can simplify. Thanks for the feedback!

The reason I went with the config store was 1. To fit with the current stateless observer component and 2. Because I think it would be good to have a reusable way to keep track of configs that plays well with mobx.

**Comment by lgeiger** [comment]
>  Because I think it would be good to have a reusable way to keep track of configs that plays well with mobx.

Thanks for explaining your train of thoughts. If you think this is generally useful, I'm happy to merge as is.

File: lib/main.js [hunk 1]
```diff
@@ -93,6 +93,12 @@ const Hydrogen = {
     );
 
     store.subscriptions.add(
+      atom.config.onDidChange("Hydrogen.statusBarDisable", ({ newValue }) => {
```
Comment (BenRussert) [inline lib/main.js]: This should be `observe` instead of `onDidChange` so it initializes. I forgot to push that change, will do so later.

**Comment by BenRussert** [comment]
I think the config store will be helpful for a number of things, for example the way we handle language mappings currently. Also anytime we use a config value within an observer component (now or in the future).

**Comment by BenRussert** [comment]
I think most people will want to leave the status bar on.

By the way here is something else we could add using the observable config map as an example:
zoom-output-font mp4

**Pull request merged**